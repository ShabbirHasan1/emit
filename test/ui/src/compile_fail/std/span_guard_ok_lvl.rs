use std::io;

#[emit::span(rt: RT, guard: span, ok_lvl: emit::Level::Info, "test")]
pub fn exec(fail: bool) -> Result<bool, io::Error> {
    if fail {
        return Err(io::Error::new(io::ErrorKind::Other, "failed"));
    }

    span.complete();

    Ok(true)
}

fn main() {

}
