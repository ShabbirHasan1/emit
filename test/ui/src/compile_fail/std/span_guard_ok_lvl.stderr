error: the `guard` control parameter is incompatible with `ok_lvl`, `err_lvl`, `panic_lvl`, or `err`
 --> src/compile_fail/std/span_guard_ok_lvl.rs:3:29
  |
3 | #[emit::span(rt: RT, guard: span, ok_lvl: emit::Level::Info, "test")]
  |                             ^^^^

warning: unused import: `std::io`
 --> src/compile_fail/std/span_guard_ok_lvl.rs:1:5
  |
1 | use std::io;
  |     ^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
